README.markdown

```diff
@@ -37,33 +23,61 @@ Download and install [CMake](http://www.cmake.org/cmake/resources/software.html)
+
 Download and unzip [ftd2xx for Windows zip file](http://www.ftdichip.com/Drivers/D2XX.htm).  Plug in a powered-up P-ROC and point the driver install wizard to the unzipped driver.  Note, this is a two-step process.  It will ask you to install a driver twice (once for USB and again for the FTDI specific stuff).
 
-Download and install [CMake](http://www.cmake.org/cmake/resources/software.html). 
+Download and install [CMake](https://cmake.org/download/). 
 
 Download and install [MinGW](http://sourceforge.net/projects/mingw/files/). (Tested with MinGW 5.1.4)
 
-Follow directions above for building yaml-cpp with the following exception:
- add '-G "MinGW Makefiles"' to the cmake command line.
+Follow directions above for Building with CMake with the following exception:
+- Add `-G "MinGW Makefiles"` to the cmake command line.
+- Use `-DEXTRA_INC="<path>"` and `-DEXTRA_LINK="<path>"` to add include/library paths for `ftd2xx.h` and `ftd2xx.sys`.
+- Use mingw32-make instead of make.
 
-To build libpinproc:
+#### Building in Windows with CMake and Visual Studio 2019
 
-- add the paths for ftd2xx.h (from the unzipped driver package) and the yaml-cpp/include/*.h files to the "include_directories" line in libpinproc/CMakeLists.txt. 
+Follow directions above for Building with CMake with the following exception:
+- Add `-A Win32` to the cmake command line.
+- Use `-DEXTRA_INC="<path>"` and `-DEXTRA_LINK="<path>"` to add include/library paths for `ftd2xx.h` and `ftd2xx.sys`.
+- Open PINPROC.sln in Visual Studio, switch to the Debug or Release configuration and perform ALL_BUILD.  It will place the libary and sample programs in `build/Debug` and `build/Release`.
 
-- either create the directory c:\usr\local\lib and copy libyaml-cpp*.a from the yaml-cpp build directory and ftd2xx.sys to it or add the location of those files to the "link_directories" line in libpinproc/CMakeLists.txt.
+Example:
+
+    cd libpinproc
+    mkdir bin; cd bin
+    cmake .. -A Win32
+    # configure paths for ftd2xxx; use `cmake .. -L` to list configured options
+    cmake .. -D EXTRA_INC="../ftd2xx"
+    cmake .. -D EXTRA_LINK="../ftd2xx/i386"
+    # Can now open PINPROC.sln in Visual Studio and build Debug and Release targets.
+
+### Testing
+
+Once built, run the `pinproctest` program with the appropriate "machine type" passed in (e.g., "wpc").  Run `pinproctest` without any parameters for a list of valid types.
 
-Follow instructions above for building libpinproc with cmake with the following exceptions:
- add '-G "MinGW Makefiles' to the cmake command line,
- use mingw32-make instead of make
- 
 ### License
 
 Copyright (c) 2009 Gerry Stellenberg, Adam Preble
 
+Copyright (c) 2020 Multimorphic, Inc.
+
+Contributors:
+  - Adam Preble
+  - Gerry Stellenberg
+  - Jan Kantert
+  - Jimmy Lipham
+  - Koen Heltzel
+  - Roy Eltham
+  - Tom Collins
+
```

Comment:
This list is based on commits on GitHub.  Sorted alphabetically by first name.  Let me know if there should be any updates to the list.